integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/MedicationFhirResourceProviderIntegrationTest.java

```diff
@@ -282,6 +328,100 @@ public void shouldUpdateExistingMedicationAsXml() throws Exception {
 		assertThat(getStrengthExtensionValue(updatedMedication), equalTo("800mg"));
 	}
 	
+	@Test
+	public void shouldReturnBadRequestWhenDocumentIdDoesNotMatchMedicationIdAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Medication/" + MEDICATION_UUID).accept(FhirMediaTypes.XML).go();
+		Medication medication = readResponse(response);
+		
+		// update the existing record
+		medication.setId(WRONG_MEDICATION_UUID);
+		
+		// send the update to the server
+		response = put("/Medication/" + MEDICATION_UUID).xmlContext(toXML(medication)).go();
+		
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldReturnNotFoundWhenUpdatingNonExistentMedicationAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Medication/" + MEDICATION_UUID).accept(FhirMediaTypes.XML).go();
+		Medication medication = readResponse(response);
+		
+		// update the existing record
+		medication.setId(WRONG_MEDICATION_UUID);
+		
+		// send the update to the server
+		response = put("/Medication/" + WRONG_MEDICATION_UUID).xmlContext(toXML(medication)).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldDeleteExistingMedicationAsJson() throws Exception {
```

Comment:
For `shouldDelete()`, we don't need the `AsJson` and `AsXML` differentiators. This is because there's no real body to either the request or the response (except in the case of an error).